network_simulator/statistics.py

```diff
@@ -34,18 +34,21 @@ class FlowStats(object):
     Contains information of a flow related to timestamps of each sending
     and receiving.
 
-    :ivar list packet_sent_times: a list of timestamps corresponding to
-    when a data packet was sent. Time is in seconds.
-    :ivar list packet_rec_times: a list of timestamps corresponding to
-    when an ACK packet was received for this :class:`.Flow`. Time is in
-    seconds.
-    :ivar list packet_rtts: a list of RTTs (in seconds).
+    :ivar list packet_sent_times: a list of (timestamp, packet_size)
+    tuples. packet_size is in bits. Time is in seconds.
+    :ivar list packet_rec_times: a list of (timestamp, packet_size)
+    tuples. packet_size is in bits. Time is in seconds.
+    :ivar list packet_rtts: a list of RTTs (timestamp, RTT) tuples where
+    RTTs are given in seconds.
+    :ivar list window_size_times: a list of (timestamp, window_size) with
+    timestamps in seconds and window_size in number of packets.
     """
     def __init__(self, packet_sent_times=[], packet_rec_times=[],
-                 packet_rtts=[]):
+                 packet_rtts=[], window_size_times=[]):
         packet_sent_times = packet_sent_times
```

Comment:
Constructor parameters need to be set via self, e.g. self.packet_sent_times = .... PyCharm should mark this as an issue in some way. This should be fixed in LinkStats and HostStats too.